packages/ui/src/models/datamapper/document.ts

```diff
@@ -100,13 +106,21 @@ export class BaseField implements IField {
   ) {
     this.id = getCamelRandomId(`field-${this.name}`, 4);
     this.path = NodePath.childOf(parent.path, this.id);
+    this.expression = name;
   }
 
   id: string;
   path: NodePath;
+  expression: string;
   fields: IField[] = [];
   isAttribute: boolean = false;
-  type = Types.AnyType;
+  protected _type = Types.AnyType;
+  public get type() {
+    return this._type;
+  }
+  public set type(value) {
+    this._type = value;
+  }
```

Comment:
[curious-question]: Why did we move from a regular property to a getter/setter combo? :thinking:

Reply:
This is a bit of dirty hack to introduce this https://github.com/KaotoIO/kaoto/pull/2298/files#diff-2ea299a35f085ca0085d0a78e21b033ee3f160f52db13fcfbe4df793e360a57eR70-R95 - when `type` is changed, update `expression` and `id` in `JsonSchemaFileld` - making XSLT3 JSON lossless representation symmetric with XML